build-tools/packages/build-infrastructure/src/versions.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as semver from "semver";
+
+import { updatePackageJsonFile } from "./packageJsonUtils.js";
+import type { IPackage, PackageJson } from "./types.js";
+
+/**
+ * Sets the version of a group of packages.
+ *
+ * Note that any loaded objects such as an IFluidRepo instance may need to be reloaded after calling this function.
+ *
+ * @param fluidRepo - The {@link IFluidRepo}.
+ * @param packages - An array of objects whose version should be updated.
+ * @param version - The version to set.
+ */
+export async function setVersion<J extends PackageJson>(
+	packages: IPackage[],
+	version: semver.SemVer,
+): Promise<void> {
+	const translatedVersion = version;
+	for (const pkg of packages) {
+		updatePackageJsonFile<J>(pkg.directory, (json) => {
```

Comment:
I think I remember this function being async...? So either this or put the promises in an array and await `Promise.all()`.
```suggestion
		await updatePackageJsonFile<J>(pkg.directory, (json) => {
```

Reply:
This function is synchronous. There's an async version I could use instead.